examples/java/ReferenceData.java

```diff
@@ -0,0 +1,89 @@
+// ReferenceData.java
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.security.KeyStore;
+import java.security.SecureRandom;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+
+public class ReferenceData {
+
+    public static void main(String[] args) {
+
+        try {
+            KeyStore clientStore = KeyStore.getInstance("PKCS12");
```

Comment:
Some brief comments explaining the various steps would help when following along with the example.